src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/GeneralizedCoastlineCheck.java

```diff
@@ -0,0 +1,111 @@
+
+    public GeneralizedCoastlineCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.percentageThreshold = this.configurationValue(configuration, "node.minimum.threshold",
+                MINIMUM_NODE_PAIR_THRESHOLD_PERCENTAGE);
+        this.minimumDistanceBetweenNodes = this.configurationValue(configuration,
+                "node.minimum.distance", MINIMUM_DISTANCE_BETWEEN_NODES);
+    }
+
+    /**
+     * This method validates or invalidates the supplied atlas object for the check
+     *
+     * @param object
+     *            The {@link AtlasObject} being checked
+     * @return True if the {@link AtlasObject} is a {@link LineItem} with the tag natural=coastline,
+     *         or has a parent relation with that tag; false otherwise
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof LineItem
+                && (Validators.isOfType(object, NaturalTag.class, NaturalTag.COASTLINE)
+                        || ((LineItem) object).relations().stream()
+                                .anyMatch((Relation relation) -> Validators.isOfType(relation,
+                                        NaturalTag.class, NaturalTag.COASTLINE)));
+    }
+
+    /**
+     * This method flags appropriate atlas objects
+     *
+     * @param object
+     *            The {@link AtlasObject} being analyzed for potential flagging
+     * @return A flag on the parameter {@link AtlasObject} if it's a generalized coastline; an empty
+     *         Optional otherwise
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final double generalizedSegments = getGeneralizedSegmentPercentage((LineItem) object);
+        if (generalizedSegments >= this.percentageThreshold)
+        {
+            return Optional.of(this.createFlag(object, this.getLocalizedInstruction(0,
+                    generalizedSegments, this.minimumDistanceBetweenNodes)));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * This method calculates the percentage of {@link Segment}s in the {@link LineItem} whose
+     * lengths are greater than or equal to the minimumDistanceBetweenNodes configuration value
+     *
+     * @param line
+     *            The LineItem whose segments are evaluated
+     * @return The percentage of generalized {@link Segment}s in the parameter {@link LineItem},
+     *         0-100 inclusive.
+     */
+    private double getGeneralizedSegmentPercentage(final LineItem line)
+    {
+        final List<Segment> segments = line.asPolyLine().segments();
+        final double innerCount = segments.stream()
+                .filter(segment -> segment.length().asMeters() >= this.minimumDistanceBetweenNodes)
+                .count();
+        return innerCount == 0.0 ? 0.0 : PERCENTAGE_ADJUST * (innerCount / segments.size());
```

Comment:
Only if segments.size() isn't 0, which might happen if a faulty LineItem without any segments but with the natural=coastline tag gets passed in

Reply:
Ok -- so really we're avoiding a divide-by-zero here? In that case, to be a little more readable could we do `segments.size() == 0 ? 0 : PERCENT_ADJUST...`? Or maybe I'm still missing what's happening

Reply:
Yes, that's the issue. innerCount would be set to 0 because the assignment statement would effectively be innerCount = <empty_stream>.count(). your suggestion is more readable so I'll make the change!